demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -632,6 +643,27 @@ static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext
                                          TRANSPORT_SEND_RECV_TIMEOUT_MS,
                                          TRANSPORT_SEND_RECV_TIMEOUT_MS );
 
+        if( opensslStatus == OPENSSL_SUCCESS )
+        {
+            /* A clean MQTT session needs to be created, if there is no session saved
+             * in this MQTT client. */
+            createCleanSession = ( *pClientSessionPresent == true ) ? false : true;
+
+            /* Sends an MQTT Connect packet using the established TLS session,
+             * then waits for connection acknowledgment (CONNACK) packet. */
+            returnStatus = establishMqttSession( pMqttContext, createCleanSession, pBrokerSessionPresent );
+
+            if( returnStatus == EXIT_FAILURE )
+            {
+                /* End TLS session, then close TCP connection. */
+                ( void ) Openssl_Disconnect( pNetworkContext );
+            }
+            else
+            {
+                *pClientSessionPresent = true;
+            }
+        }
+
         if( opensslStatus != OPENSSL_SUCCESS )
```

Comment:
The MQTT connection changes `returnStatus` but this only tests `opensslStatus`
```suggestion
        if( ( opensslStatus != OPENSSL_SUCCESS ) || ( returnStatus != EXIT_SUCCESS ) )
```